reana_cluster/cli/cluster.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2017 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""REANA cluster administration (init, restart, etc.) commands."""
+
+
+import logging
+
+import click
+import os
+
+import errno
+import yaml
+
+from ..config import generated_cluster_conf_default_path
+
+from reana_cluster import KubernetesBackend
+
+from ..utils import load_spec_file
+
+
+@click.command()
+@click.option('--remove-persistent-storage', is_flag=True,
+              help='If set, also persistent storage '
+                   'inside the cluster is deleted.')
+@click.pass_context
+def down(ctx, remove_persistent_storage):
+    """Bring REANA cluster down, i.e. shutdown all components."""
+
+    try:
+        cluster_spec = load_spec_file(click.format_filename(
+            ctx.parent.params['file']),
+            ctx.parent.params['skip_validation'])
+
+        backend = KubernetesBackend(cluster_spec)
```

Comment:
even `backend` could be moved up there if we identify that all commands need a backend object, but I am not sure, what do you think?

Reply:
Hmm. The previous comments on spec file parsing serve a purpose here, but I have to admit that this sounds like a good idea. I will check if all current (and future) functionality depend on a backend.